src/android/Geolocation.java

```diff
@@ -128,50 +193,58 @@ public void onRequestPermissionResult(int requestCode, String[] permissions, int
         }
     }
 
-    private boolean checkGooglePlayServicesAvailable(CallbackContext callbackContext) {
-        GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
-        int status = googleApiAvailability.isGooglePlayServicesAvailable(cordova.getActivity());
+    private void getLocationGms(LocationContext locationContext) {
```

Comment:
Hi, first of all, thanks for your point, but in this case, we cannot improve the code, because inside each function I know that looks identical, but if you see inside function gms the LocationRequest is instantiate from google package and inside hms code, the LocationRequest is coming from huawei class, in this case we cannot reuse the same code to create 2 differents objects to pass the requestLocationUpdatesIfSettingsSatisfied.. something.

But thanks for your point of view, by the way.

Reply:
Actually, I have to disagree with you and I'll explain:

- The 3 local variables set before `request` (`args`, `timeout` and `enableHighAccuracy`) are actually the same.
- I'll explain a way you could do what I mentioned: you could create an interface with methods that are "shared" by both `LocationRequest` implementations. You could then extend both implementations to extend the new interface and, instead of using the concrete implementation on the methods, you could use the new interface and have only 1 logic to satisfy both scenarios (Dependency Inversion Principle).

Still, feel free to keep the code as it is and resolve this conversations.